diff -rc nix-1.11.4/src/libstore/build.cc nix-1.11.4pre1234_abcdef/src/libstore/build.cc
*** nix-1.11.4/src/libstore/build.cc	1970-01-01 00:00:01.000000000 +0000
--- nix-1.11.4pre1234_abcdef/src/libstore/build.cc	1970-01-01 00:00:01.000000000 +0000
***************
*** 761,767 ****
      GoalState state;
  
      /* Stuff we need to pass to initChild(). */
!     typedef map<Path, Path> DirsInChroot; // maps target path to source path
      DirsInChroot dirsInChroot;
      typedef map<string, string> Environment;
      Environment env;
--- 761,774 ----
      GoalState state;
  
      /* Stuff we need to pass to initChild(). */
!     struct ChrootPath {
!         Path source;
!         bool optional;
!         ChrootPath(Path source = "", bool optional = false)
!             : source(source), optional(optional)
!         { }
!     };
!     typedef map<Path, ChrootPath> DirsInChroot; // maps target path to source path
      DirsInChroot dirsInChroot;
      typedef map<string, string> Environment;
      Environment env;
***************
*** 1924,1943 ****
  
          dirsInChroot.clear();
  
!         for (auto & i : dirs) {
              size_t p = i.find('=');
              if (p == string::npos)
!                 dirsInChroot[i] = i;
              else
!                 dirsInChroot[string(i, 0, p)] = string(i, p + 1);
          }
          dirsInChroot[tmpDirInSandbox] = tmpDir;
  
          /* Add the closure of store paths to the chroot. */
          PathSet closure;
          for (auto & i : dirsInChroot)
!             if (isInStore(i.second))
!                 computeFSClosure(worker.store, toStorePath(i.second), closure);
          for (auto & i : closure)
              dirsInChroot[i] = i;
  
--- 1931,1960 ----
  
          dirsInChroot.clear();
  
!         for (auto i : dirs) {
!             if (i.empty()) continue;
!             bool optional = false;
!             if (i[i.size() - 1] == '?') {
!                 optional = true;
!                 i.pop_back();
!             }
              size_t p = i.find('=');
              if (p == string::npos)
!                 dirsInChroot[i] = {i, optional};
              else
!                 dirsInChroot[string(i, 0, p)] = {string(i, p + 1), optional};
          }
          dirsInChroot[tmpDirInSandbox] = tmpDir;
  
          /* Add the closure of store paths to the chroot. */
          PathSet closure;
          for (auto & i : dirsInChroot)
!             try {
!                 if (isInStore(i.second.source))
!                     computeFSClosure(worker.store, toStorePath(i.second.source), closure);
!             } catch (Error & e) {
!                 throw Error(format("while processing ‘build-sandbox-paths’: %s") % e.what());
!             }
          for (auto & i : closure)
              dirsInChroot[i] = i;
  
***************
*** 2328,2339 ****
                 environment. */
              for (auto & i : dirsInChroot) {
                  struct stat st;
!                 Path source = i.second;
                  Path target = chrootRootDir + i.first;
                  if (source == "/proc") continue; // backwards compatibility
                  debug(format("bind mounting ‘%1%’ to ‘%2%’") % source % target);
!                 if (stat(source.c_str(), &st) == -1)
!                     throw SysError(format("getting attributes of path ‘%1%’") % source);
                  if (S_ISDIR(st.st_mode))
                      createDirs(target);
                  else {
--- 2345,2360 ----
                 environment. */
              for (auto & i : dirsInChroot) {
                  struct stat st;
!                 Path source = i.second.source;
                  Path target = chrootRootDir + i.first;
                  if (source == "/proc") continue; // backwards compatibility
                  debug(format("bind mounting ‘%1%’ to ‘%2%’") % source % target);
!                 if (stat(source.c_str(), &st) == -1) {
!                     if (i.second.optional && errno == ENOENT)
!                         continue;
!                     else
!                         throw SysError(format("getting attributes of path ‘%1%’") % source);
!                 }
                  if (S_ISDIR(st.st_mode))
                      createDirs(target);
                  else {
***************
*** 2519,2533 ****
               */
              sandboxProfile += "(allow file-read* file-write* process-exec\n";
              for (auto & i : dirsInChroot) {
!                 if (i.first != i.second)
                      throw Error(format(
                          "can't map '%1%' to '%2%': mismatched impure paths not supported on Darwin")
!                         % i.first % i.second);
  
                  string path = i.first;
                  struct stat st;
!                 if (lstat(path.c_str(), &st))
                      throw SysError(format("getting attributes of path ‘%1%’") % path);
                  if (S_ISDIR(st.st_mode))
                      sandboxProfile += (format("\t(subpath \"%1%\")\n") % path).str();
                  else
--- 2540,2557 ----
               */
              sandboxProfile += "(allow file-read* file-write* process-exec\n";
              for (auto & i : dirsInChroot) {
!                 if (i.first != i.second.source)
                      throw Error(format(
                          "can't map '%1%' to '%2%': mismatched impure paths not supported on Darwin")
!                         % i.first % i.second.source);
  
                  string path = i.first;
                  struct stat st;
!                 if (lstat(path.c_str(), &st)) {
!                     if (i.second.optional && errno == ENOENT)
!                         continue;
                      throw SysError(format("getting attributes of path ‘%1%’") % path);
+                 }
                  if (S_ISDIR(st.st_mode))
                      sandboxProfile += (format("\t(subpath \"%1%\")\n") % path).str();
                  else
diff -rc nix-1.11.4/src/libstore/globals.cc nix-1.11.4pre1234_abcdef/src/libstore/globals.cc
*** nix-1.11.4/src/libstore/globals.cc	1970-01-01 00:00:01.000000000 +0000
--- nix-1.11.4pre1234_abcdef/src/libstore/globals.cc	1970-01-01 00:00:01.000000000 +0000
***************
*** 188,193 ****
--- 188,195 ----
      _get(enableImportNative, "allow-unsafe-native-code-during-evaluation");
      _get(useCaseHack, "use-case-hack");
      _get(preBuildHook, "pre-build-hook");
+     _get(keepGoing, "keep-going");
+     _get(keepFailed, "keep-failed");
  
      string subs = getEnv("NIX_SUBSTITUTERS", "default");
      if (subs == "default") {
