diff -rc nix-1.11.2/src/libstore/build.cc nix-1.11.2-new/src/libstore/build.cc
*** nix-1.11.2/src/libstore/build.cc	2016-02-11 09:12:17.000000000 +0000
--- nix-1.11.2-new/src/libstore/build.cc	2016-09-21 13:34:40.318572190 +0000
***************
*** 2349,2355 ****
  
              /* Mount a new tmpfs on /dev/shm to ensure that whatever
                 the builder puts in /dev/shm is cleaned up automatically. */
!             if (pathExists("/dev/shm") && mount("none", (chrootRootDir + "/dev/shm").c_str(), "tmpfs", 0, 0) == -1)
                  throw SysError("mounting /dev/shm");
  
              /* Mount a new devpts on /dev/pts.  Note that this
--- 2349,2355 ----
  
              /* Mount a new tmpfs on /dev/shm to ensure that whatever
                 the builder puts in /dev/shm is cleaned up automatically. */
!             if (pathExists("/dev/shm") && mount("none", (chrootRootDir + "/dev/shm").c_str(), "tmpfs", 0, "size=75%") == -1)
                  throw SysError("mounting /dev/shm");
  
              /* Mount a new devpts on /dev/pts.  Note that this
